spec/unit/etc_spec.rb

```diff
@@ -0,0 +1,155 @@
+require 'puppet/etc'
+require 'spec_helper'
+
+# The Ruby::Etc module is largely non-functional on Windows - many methods
+# simply return nil regardless of input, the Etc::Group struct is not defined,
+# and Etc::Passwd is missing fields
+describe Puppet::Etc, :if => !Puppet.features.microsoft_windows? do
```

Comment:
IMO it's reasonable to surface the underlying behavior of `Etc` on Windows; in an ideal world the underlying code would raise a `NotImplementedError` if invoked on Windows but adding that here would be changing the contract. :+1: for this as it is.